pkg/server/log.go

```diff
@@ -1,77 +1,50 @@
 package server
 
 import (
-	"bytes"
-	"net/http"
+	"context"
 	"time"
 
 	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/peer"
 )
 
-type loggingResponseWriter struct {
-	http.ResponseWriter
-	statusCode int
-	buf        *bytes.Buffer
-}
-
-func NewLoggingResponseWriter(w http.ResponseWriter) *loggingResponseWriter {
-	return &loggingResponseWriter{
-		w,
-		http.StatusOK,
-		&bytes.Buffer{},
-	}
-}
-
-func (lrw *loggingResponseWriter) WriteHeader(code int) {
-	lrw.statusCode = code
-	lrw.ResponseWriter.WriteHeader(code)
-}
-
-func (lrw *loggingResponseWriter) Write(b []byte) (int, error) {
-	return lrw.buf.Write(b)
-}
+func logInterceptor(logger *zap.Logger) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		start := time.Now()
 
-func (lrw *loggingResponseWriter) Flush() {
-	if f, ok := lrw.ResponseWriter.(http.Flusher); ok {
-		f.Flush()
+		// Call the handler to execute the gRPC request
+		response, err := handler(ctx, req)
+
+		duration := time.Since(start).Milliseconds()
+		fields := []zap.Field{
+			zap.Int64("duration_ms", duration),
+			zap.String("method", info.FullMethod),
+			zap.Any("request", req),
+		}
+		// Get the remote address from the context
+		peer, ok := peer.FromContext(ctx)
+		if ok {
+			fields = append(fields, zap.String("remote_addr", peer.Addr.String()))
+		}
+
+		if err != nil {
+			fields = append(fields, zap.Int32("status_code", int32(grpc.Code(err))))
```

Comment:
I saw that `grpc.Code` is deprecated and it suggests using `status.Code` instead. https://pkg.go.dev/google.golang.org/grpc@v1.54.0#Code